src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocLeadingAsteriskAlignCheck.java

```diff
@@ -0,0 +1,246 @@
+ * Default value is {@code false}.
+ * </li>
+ * </ul>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code javadoc.asterisk.indentation}
+ * </li>
+ * <li>
+ * {@code javadoc.missed.html.close}
+ * </li>
+ * <li>
+ * {@code javadoc.parse.rule.error}
+ * </li>
+ * <li>
+ * {@code javadoc.unclosedHtml}
+ * </li>
+ * <li>
+ * {@code javadoc.wrong.singleton.html.tag}
+ * </li>
+ * </ul>
+ *
+ * @since 10.17.0
+ */
+@GlobalStatefulCheck
+public class JavadocLeadingAsteriskAlignCheck extends AbstractJavadocCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "javadoc.asterisk.indentation";
+
+    /** Specifies the line number of starting block of the javadoc comment. */
+    private int javadocStartLineNumber;
+
+    /** Specifies the column number of starting block of the javadoc comment with tabs expanded. */
+    private int expectedColumnNumberTabsExpanded;
+
+    /**
+     * Specifies the column number of the leading asterisk
+     * without tabs expanded.
+     */
+    private int expectedColumnNumberWithoutExpandedTabs;
+
+    /** Specifies the lines of the file being processed. */
+    private String[] fileLines;
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.LEADING_ASTERISK,
+        };
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return getAcceptableJavadocTokens();
+    }
+
+    @Override
+    public void beginJavadocTree(DetailNode rootAst) {
+        // this method processes and sets information of starting javadoc tag.
+        fileLines = getLines();
+        final String startLine = fileLines[rootAst.getLineNumber() - 1];
+        javadocStartLineNumber = rootAst.getLineNumber();
+        expectedColumnNumberTabsExpanded = CommonUtil.lengthExpandedTabs(
+            startLine, rootAst.getColumnNumber() - 1, getTabWidth());
+    }
+
+    @Override
+    public void visitJavadocToken(DetailNode ast) {
+        // this method checks the alignment of leading asterisks.
+        final boolean isJavadocStartingLine = ast.getLineNumber() == javadocStartLineNumber;
+
+        if (!isJavadocStartingLine) {
+            final OptionalInt leadingAsteriskColumnNumber =
+                                        getAsteriskColumnNumber(ast.getText());
+
+            leadingAsteriskColumnNumber
+                    .stream()
+                    .map(columnNumber -> expandedTabs(ast.getText(), columnNumber))
+                    .filter(expandedColumnNumber -> {
+                        return hasInvalidAlignment(
+                                expectedColumnNumberTabsExpanded,
+                                expandedColumnNumber);
+                    })
+                    .forEach(violationColumnNumber -> {
```

Comment:
How can we have more than one at a time?

Reply:
https://github.com/checkstyle/checkstyle/pull/14643#discussion_r1614833279

@Zopsss Ok, I see that OptionalInt does not behave similarly to the boxed version. If we make some usage of APIs that OptionalInt does not support, will CI still complain?

Reply:
> If we make some usage of APIs that OptionalInt does not support, will CI still complain?

I don't know, It's my first time working with OptionalInt

> This should absolutely be ifPresent, we don't need to use streams here.

New approach using ifPresent:

https://github.com/checkstyle/checkstyle/blob/4c812934107632c64377b91dc685c1a082c02fce/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocLeadingAsteriskAlignCheck.java#L139-L149